spec/spec_helper.rb

```diff
@@ -1,4 +1,20 @@
-require 'capistrano/all'
-
 $LOAD_PATH.unshift File.expand_path('../../lib', __FILE__)
-require 'capistrano/bundle_rsync'
+
+# Emulate cap command
+# https://github.com/capistrano/capistrano/blob/31e142d56f8d894f28404fb225dcdbe7539bda18/bin/cap
+require "capistrano/all"
+
+# Emulate Capfile
+
+# Load DSL and Setup Up Stages
+require 'capistrano/setup'
+
+# Includes default deployment tasks
+require 'capistrano/deploy'
+
+if Gem::Version.new(Capistrano::VERSION) >= Gem::Version.new('3.7')
+  require 'capistrano/bundle_rsync/plugin'
+  install_plugin Capistrano::BundleRsync::Plugin
+else
+  require 'capistrano/bundle_rsync'
+end
```

Comment:
This is almost the same Capfile as that in README.md

https://github.com/sonots/capistrano-bundle_rsync/blob/811964f5f79cbecb5a8cc84b7db4ebc0e297e36f/README.md#L142-L156

The differences are as follows:

* conditional require for Plugin style or good old style.
* Skip `rbenv` and loading tasks

The reason why the previous test was failure is that `Defaults.set_defaults` is evaluated without requiring Capistrano DSLs. I have fixed the issue by simulating Capistrano loading flow in the spec_helper.